osquery/tables/system/linux/os_version.cpp

```diff
@@ -83,17 +86,26 @@ QueryData genOSRelease() {
 }
 
 QueryData genOSVersion(QueryContext& context) {
-  if (isReadable(kOSRelease)) {
+  // Funtoo has an empty os-release file
+  if (isReadable(kOSRelease) && boost::filesystem::file_size(kOSRelease)) {
```

Comment:
Ouch, this is an odd scenario.

For this method, I'd recommend something like:
```cpp
QueryData genOSVersion(QueryContext& context) {
  if (isReadable(kOSRelease)) {
    boost::system::error_code ec;
    // Funtoo has an empty os-release file.
    if (boost::filesystem::file_size(kOSRelease, ec) > 0) {
      return genOSRelease();
    }
  }

  Row r;
  std::string content;
  if (readFile(kRedhatRelease, content).ok()) {
    r["platform_like"] = "rhel";
  } else if (readFile(kGentooRelease, content).ok()) {
    r["platform_like"] = "gentoo";
  } else {
    return {};
  }

  boost::algorithm::trim_all(content);

  [...]
```